src/app/zap-templates/zcl/data-model/test/mode-select-extensions.xml

```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0"?>
+<!--
+Copyright (c) 2022 Project CHIP Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<configurator>
+  <domain name="CHIP"/>
+  <!-- This extends the ModeSelect cluster -->
+  <clusterExtension code="0x0050">
+    <!--
+       Manufacturer specific extension attribute:
+          - Prefix: 0xFFF1 - Test vendor MC
+          - Suffix: 0x0001 - Non-global attribute id
+     -->
+    <attribute side="server" code="0xFFF10001" define="MEI_EXTENSION" type="INT8U" writable="false" optional="true" isNullable="false">ManufacturerExtension</attribute>
+
+    <!--
+       Manufacturer specific extension command:
+          - Prefix: 0xFFF1 - Test vendor MC
+          - Suffix: 0x0001 - command id
+     -->
+    <command source="client" code="0xFFF10001" name="SampleMfgExtensionCommand" optional="true">
```

Comment:
It should, but it failed to compile when I enabled it in the zap file, so for now I defined the extension but it is not activated because it does not compile.

I would make it compile as a separate PR if we ever get to it.

Reply:
It failed to compile because the codegen assumes that the app-common codegen knows all the XML bits, and this PR is not changing app-common to use this new bit...

Reply:
That is a problem though: I explicitly want all-clusters-app to have some capabilities that NONE of the other applications know about.

This is the same as a vendor using the SDK to compile their own custom app. It seems like for this the SDK itself needs to change, which seems like a bug to me.

Reply:
Given complexity I doubt this will be supported for 1.0 ... at least we will have attribute override supported. I also expect vendors to likely create "custom cluster" but less likely to want "extend standard cluster with new commands/attributes".

Reply:
So, if I understand this PR correctly, extending attributes works, but not commands/events? If so, can we file an issue to track that specifically?